### PR TITLE
Introduction of month span

### PR DESCRIPTION
**Pull request by Corniel**
Basically a subset of the date span, without day precision.

### CODE REVIEW

File: src/Qowaiv/Date.cs [hunk 1]
```diff
@@ -426,6 +442,12 @@ public string ToString(string format, IFormatProvider formatProvider)
         /// <summary>Subtracts the Time Span from the date.</summary>
         public static Date operator -(Date d, TimeSpan t) => d.Subtract(t);
 
+        /// <summary>Adds the month span to the date.</summary>
+        public static Date operator +(Date date, MonthSpan span) => date.Add(span);
+
+        /// <summary>Subtracts the month span to the date.</summary>
```
Comment (LauraXiulan) [inline src/Qowaiv/Date.cs:448]: **from** the date.

File: src/Qowaiv/DateSpan.cs [hunk 2]
```diff
@@ -278,13 +278,13 @@ public string ToString(string format, IFormatProvider formatProvider)
         /// <returns>
         /// Returns a date span describing the duration between t1 and t2.
         /// </returns>
-        public static DateSpan Subtract(Date t1, Date t2) => Subtract(t1, t2, DateSpanSettings.Default);
+        public static DateSpan Subtract(Date d1, Date t2) => Subtract(d1, t2, DateSpanSettings.Default);
```
Comment (LauraXiulan) [inline src/Qowaiv/DateSpan.cs:281]: Shouldn't it be d1, **d2** for consistency

File: src/Qowaiv/DateSpan.cs [hunk 3]
```diff
@@ -268,8 +268,8 @@ public string ToString(string format, IFormatProvider formatProvider)
         /// </returns>
         public static DateSpan Age(Date t1, Date reference) => Subtract(reference, t1, DateSpanSettings.WithoutMonths);
 
-        /// <summary>Creates a date span on by subtracting t2 from t1.</summary>
-        /// <param name="t1">
+        /// <summary>Creates a date span on by subtracting <paramref name="d1"/> from <paramref name="t2"/>.</summary>
```
Comment (LauraXiulan) [inline src/Qowaiv/DateSpan.cs:271]: remove 'on'

File: src/Qowaiv/DateSpan.cs [hunk 4]
```diff
@@ -278,13 +278,13 @@ public string ToString(string format, IFormatProvider formatProvider)
         /// <returns>
         /// Returns a date span describing the duration between t1 and t2.
```
Comment (LauraXiulan) [inline src/Qowaiv/DateSpan.cs:279]: between **d1** and t2.

File: src/Qowaiv/DateSpan.cs [hunk 5]
```diff
@@ -293,30 +293,30 @@ public string ToString(string format, IFormatProvider formatProvider)
...
```
Comment (LauraXiulan) [inline src/Qowaiv/DateSpan.cs:294]: between **d1 and d2**.

File: src/Qowaiv/DateSpan.cs [hunk 6]
```diff
@@ -278,13 +278,13 @@ public string ToString(string format, IFormatProvider formatProvider)
...
         /// </returns>
-        public static DateSpan Subtract(Date t1, Date t2) => Subtract(t1, t2, DateSpanSettings.Default);
+        public static DateSpan Subtract(Date d1, Date t2) => Subtract(d1, t2, DateSpanSettings.Default);
 
-        /// <summary>Creates a date span on by subtracting t2 from t1.</summary>
-        /// <param name="t1">
+        /// <summary>Creates a date span on by subtracting <paramref name="d1"/> from <paramref name="d2"/>.</summary>
```
Comment (LauraXiulan) [inline src/Qowaiv/DateSpan.cs:283]: remove 'on'

File: test/Qowaiv.UnitTests/LocalDateTimeTest.cs [hunk 7]
```diff
@@ -643,6 +643,25 @@ public void AddMonths_12_AreEqual()
...
+            var act = TestStruct + MonthSpan.FromYears(1);
+            var exp = new LocalDateTime(1989, 06, 13, 22, 10, 05, 001);
+
+            Assert.AreEqual(exp, act);
+        }
+
+        [Test]
+        public void Subtract_1Month_AreEqual()
+        {
+            var act = TestStruct + MonthSpan.FromYears(1);
```
Comment (LauraXiulan) [inline test/Qowaiv.UnitTests/LocalDateTimeTest.cs:659]: TestStruct - MonthSpan.FromMonths(1)

File: test/Qowaiv.UnitTests/DateTest.cs [hunk 8]
```diff
@@ -831,6 +831,21 @@ public void AddMonths_12_AreEqual()
...
+
+        [Test]
+        public void Add_12Months_AreEqual()
+        {
+            var added = new Date(1970, 02, 14) + MonthSpan.FromMonths(12);
+            Assert.AreEqual(new Date(1971, 02, 14), added);
+        }
+
+        [Test]
+        public void Subtract_12Months_AreEqual()
```
Comment (LauraXiulan) [inline test/Qowaiv.UnitTests/DateTest.cs:843]: Subtract_**3**Months_AreEqual

File: src/Qowaiv/LocalDateTime.cs [hunk 9]
```diff
@@ -536,6 +553,12 @@ public string ToString(string format, IFormatProvider formatProvider)
         /// <summary>Subtracts the Time Span from the local date time.</summary>
         public static LocalDateTime operator -(LocalDateTime d, TimeSpan t) => d.Subtract(t);
 
+        /// <summary>Adds the month span to the date.</summary>
+        public static LocalDateTime operator +(LocalDateTime date, MonthSpan span) => date.Add(span);
+
+        /// <summary>Subtracts the month span to the date.</summary>
```
Comment (LauraXiulan) [inline src/Qowaiv/LocalDateTime.cs:559]: **from** the date.

File: test/Qowaiv.UnitTests/MonthSpanTest.cs [hunk 10]
```diff
@@ -0,0 +1,701 @@
...
+        /// <summary>MonthSpan.Zero should be equal to the default of month span.</summary>
+        [Test]
+        public void Zero_EqualsDefault()
+        {
+            Assert.AreEqual(default(MonthSpan), MonthSpan.Zero);
+        }
+
+        /// <summary>TryParse null should be valid.</summary>
+        [Test]
+        public void TyrParse_Null_IsValid()
```
Comment (LauraXiulan) [inline test/Qowaiv.UnitTests/MonthSpanTest.cs:28]: **Try**Parse

File: src/Qowaiv/MonthSpan.cs [hunk 11]
```diff
@@ -0,0 +1,356 @@
...
+        /// <param name = "format">
+        /// The format that this describes the formatting.
+        /// </param>
+        /// <param name = "formatProvider">
+        /// The format provider.
+        /// </param>
+        /// <remarks>
+        /// The formats:
+        /// 
+        /// F: as 0Y+0D.
```
Comment (LauraXiulan) [inline src/Qowaiv/MonthSpan.cs:187]: 0D or 0M?

File: src/Qowaiv/MonthSpan.cs [hunk 12]
```diff
@@ -0,0 +1,356 @@
...
+        public static MonthSpan FromMonths(int months)
+        {
+            if (TryCreate(months, out var monthSpan))
+            {
+                return monthSpan;
+            }
+            throw new ArgumentOutOfRangeException(nameof(months), QowaivMessages.FormatExceptionMonthSpan);
+        }
+
+        /// <summary>Creates a month span on by subtracting <paramref name="d1"/> from <paramref name="d2"/>.</summary>
```
Comment (LauraXiulan) [inline src/Qowaiv/MonthSpan.cs:304]: Remove 'on'

File: src/Qowaiv/MonthSpan.cs [hunk 13]
```diff
@@ -0,0 +1,356 @@
...
+        /// <param name="d1">
+        /// The date to subtract from.
+        /// </param>
+        /// <param name="d2">
+        /// The date to subtract.
+        /// </param>
+        /// <returns>
+        /// Returns a month span describing the duration between t1 and t2.
```
Comment (LauraXiulan) [inline src/Qowaiv/MonthSpan.cs:312]: d1 and d2

File: test/Qowaiv.UnitTests/MonthSpanTest.cs (hunk 10 above)
Comment (Corniel) [reply test/Qowaiv.UnitTests/MonthSpanTest.cs]: Oops, 100+ occurences. ;)

**Pull request merged**